src/pcre2_compile_class.c

```diff
@@ -372,9 +372,9 @@ while (*ptr != META_CLASS_END)
 #ifdef SUPPORT_UNICODE
     if (options & PARSE_CLASS_CASELESS_UTF)
       {
-      size_t usize = utf_caseless_extend(start_char, *ptr++, options, buffer);
-      if (buffer != NULL) buffer += usize;
-      total_size += usize;
+      size = utf_caseless_extend(start_char, *ptr++, options, buffer);
```

Comment:
FYI this will revert Phillip's last fix for `-Wshadow`, is that intended?

Reply:
Yes. I wanted to remove the `size_t` but forgot it.

Reply:
nevermind, shold had pulled it first before commenting, guess that is why it was using `size` in the original to begin with then ;)

Reply:
Oh no. I found another issue with negated ascii classes. And this is still just the range merge, the logarithmic search is still very far.